src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -221,13 +218,8 @@ protected Future<SchemaRegistryStatus> getSchemaRegistryConnectionStatus() {
               ).build()
       );
     }).orElseGet(() -> {
-      // There is no Schema Registry configuration, so return the NONE state
-      return Future.succeededFuture(
-          ConnectionStatusSchemaRegistryStatusBuilder
-              .builder()
-              .state(ConnectedState.NONE)
-              .build()
-          );
+      // There is no Schema Registry configuration, so return no status for SR
+      return Future.succeededFuture(null);
```

Comment:
This is the bug I found and fixed: when there is no `spec.schema_registry`, then we want `status.schema_registry` to also be null.